### PR TITLE
Create CONTRIBUTING.md

### LINKED ISSUE #1382: Contributing guide missing
**Describe the bug**
`README.md` links to a Contributing Guide over at https://github.com/open-telemetry/opentelemetry-lambda/blob/main/CONTRIBUTING.md, whic is missing

**Steps to reproduce**
Click on the `Contributing Guide` link in the readme: https://github.com/open-telemetry/opentelemetry-lambda/blob/10cc2ef61b9ac618ee7793496e61811d03c06e68/README.md?plain=1#L104

**What did you expect to see?**
The contributing guide.

**What did you see instead?**
404 - File not found

### PR DESCRIPTION
**Pull request by tylerbenson**
Fixes #1382

(Based on https://github.com/open-telemetry/sig-security/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# OpenTelemetry Contributor Guide
+
+Welcome to OpenTelemetry! This document is the single source of truth for how to
```
Comment (nslaughter) [inline CONTRIBUTING.md]:
```suggestion
Welcome to OpenTelemetry! This document provides general guidance for contributing to 
this codebase. This is a multi-lingual codebase, so sub-directories like `collector` may
provide additional guidance in a CONTRIBUTING.md.
```

The CONTRIBUTING.md in `opentelemetry-collector-contrib` has a lot more prescriptive detail which I think is quite valuable. The trouble is that they get into dev workflows and we probably shouldn't do that in the top-level document.

My thought in making this change in the language is that we could start by adding a CONTRIBUTING.md to the `collector` when time allows and follow up with additional consultation. I've noticed with PRs in the Collector that guidelines in that CONTRIBUTING.md could improve reviews and code quality.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+Conduct](https://github.com/open-telemetry/community/blob/master/code-of-conduct.md).
+
+### Sign the CLA
+
+Before you can contribute, you will need to sign the [Contributor License
+Agreement](https://docs.linuxfoundation.org/lfx/easycla/contributors).
+
+### Code attribution
+
+[License information](README.md#License) should be included in all source files where applicable.
```
Comment (nslaughter) [inline CONTRIBUTING.md:24]: There's a workflow in the Makefile for this.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+[OSI-approved licenses](https://opensource.org/licenses/alphabetical) known to be compatible.
+
+If you make substantial changes to the third-party code, _prepend_ the contributed
+third party file with OpenTelemetry's copyright notice.
+
+If the contributed code is not third-party code and you are the author we
+strongly encourage to avoid including your name in the notice and use the
+generic "Copyright The OpenTelemetry Authors" notice. See rationale for this
+recommendation [here](https://github.com/cncf/foundation/blob/master/copyright-notices.md#why-not-list-every-copyright-holder).
+
```
Comment (nslaughter) [inline CONTRIBUTING.md:49]: Some much needed clarity for just about everybody. Glad we're including this in guidance.

**Review by nslaughter — COMMENTED** [review]
I think this is good. I have two basic suggestions:
(1) see my comments re: different CONTRIBUTING.md files for some sub-directories. I think there's a good opportunity to make workflows more straightforward, but it needs to be specialized in the sub-directories and priotized
(2) the [CONTRIBUTING.md in `opentelemetry-collector-contrib`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#issue-triaging) is explicit about issue triaging workflows.

**Review by rapphil — APPROVED** [review]

**Pull request merged**